### PR TITLE
feat: Kube-Flux - Added features of carml contribution

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

> This is a follow up to: https://github.com/Azure/ResourceModules/pull/4142

<details>
<summary>The current version of flux configuration 2022-03-01 doesn't add Postbuild substitution variables</summary>
Updated Microsoft.KubernetesConfiguration/fluxConfigurations to version 2023-05-01
</details>

> Please include a summary of the change and which issue is fixed.

Updated Microsoft.KubernetesConfiguration/fluxConfigurations to version 2023-05-01 and added postBuild section in common  main.test.bicep.

> Please also include the context.

The current version 2022-03-01 doesn't add Postbuild substitution variables which are included in the Bicep configuration.  We have tested these by running the common/main.test.bicep with 2 postBuild substitute variables with values foo and bar and they don't appear in the Kustomization in AKS.  Please see snippet below which is a result of running 'kubectl describe kustomization fluxaakcfccom001-unified -n flux-system'

image

We have updated the API version to 2023-05-01 which renders the expected results.  Please see snippet below which is a result of running 'kubectl describe kustomization fluxaakcfccom001-unified -n flux-system'

image

This is a feature we require to correctly configure our Flux configuration.

| Pipeline |
| - |
| [avm.res.kubernetes-configuration.flux-configuration](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.flux-configuration.yml) |
| [avm.res.kubernetes-configuration.extension](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.extension.yml) |

There are no dependencies

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: avm/res/kubernetes-configuration/flux-configuration/main.bicep [hunk 1]
```diff
@@ -24,7 +24,7 @@ param configurationProtectedSettings object?
 @description('Conditional. Parameters to reconcile to the GitRepository source kind type. Required if `sourceKind` is `GitRepository`.')
 param gitRepository object?
 
-@description('Optional. Array of kustomizations used to reconcile the artifact pulled by the source type on the cluster.')
+@description('Required. Array of kustomizations used to reconcile the artifact pulled by the source type on the cluster.')
```
Comment (eriqua) [inline avm/res/kubernetes-configuration/flux-configuration/main.bicep:27]: This should require a minor version bump (breaking change until v1.0.0 is out)

Comment (AlexanderSehr) [reply avm/res/kubernetes-configuration/flux-configuration/main.bicep:27]: True true

File: avm/res/kubernetes-configuration/extension/main.bicep [hunk 2]
```diff
@@ -94,7 +94,7 @@ module fluxConfiguration 'br/public:avm/res/kubernetes-configuration/flux-config
     bucket: fluxConfiguration.?bucket
     configurationProtectedSettings: fluxConfiguration.?configurationProtectedSettings
     gitRepository: fluxConfiguration.?gitRepository
-    kustomizations: fluxConfiguration.?kustomizations
+    kustomizations: fluxConfiguration.kustomizations
```
Comment (eriqua) [inline avm/res/kubernetes-configuration/extension/main.bicep:97]: I think we should avoid releasing kube extensions in this PR. I'd suggest to:
Update kube config and release 0.3.0 in this PR
Update kube extensions in PR #613

Comment (AlexanderSehr) [reply avm/res/kubernetes-configuration/extension/main.bicep]: Done @eriqua

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**